WootricSDK/WootricSDK/WTRiPADSurveyViewController.m

```diff
@@ -165,12 +165,39 @@ - (void)dismissViewControllerWithBackgroundFade {
 }
 
 - (void)facebookButtonPressed {
-  NSURL *url = _settings.facebookPage;
-  if (![[UIApplication sharedApplication] openURL:url]) {
-    NSLog(@"Failed to open facebook page");
+  if ([SLComposeViewController isAvailableForServiceType:SLServiceTypeFacebook]) {
+    SLComposeViewController *facebookSheet = [SLComposeViewController composeViewControllerForServiceType:SLServiceTypeFacebook];
+    [facebookSheet addURL:_settings.facebookPage];
+    [facebookSheet setCompletionHandler:^(SLComposeViewControllerResult result){
+      switch (result) {
+        case SLComposeViewControllerResultCancelled:
+          NSLog(@"WootricSDK: Post cancelled");
+          break;
+        case SLComposeViewControllerResultDone:
+          NSLog(@"WootricSDK: Post successful");
+          break;
+        default:
+          break;
+      }
+    }];
+    [self presentViewController:facebookSheet animated:YES completion:nil];
+  } else {
+    UIAlertView *alertView = [[UIAlertView alloc] initWithTitle:@"Sorry"
+                                                        message:@"You can't send a tweet right now, make sure your device has an internet connection and you have at least one Facebook account setup"
+                                                       delegate:self
+                                              cancelButtonTitle:@"OK"
+                                              otherButtonTitles:nil];
+    [alertView show];
   }
 }
 
+- (void)facebookLikeButtonPressed {
+    NSURL *url = _settings.facebookPage;
+    if (![[UIApplication sharedApplication] openURL:url]) {
+        NSLog(@"Failed to open facebook page");
+    }
+}
+
 - (void)twitterButtonPressed {
```

Comment:
I assume the code for ``` -(void) twitterButtonPressed ``` is almost the same as ``` -(void) facebookButtonPressed ```, if that's the case, you could refactor them and move the common logic to a single method.